src/WebJobs.Script.WebHost/Handlers/SystemTraceHandler.cs

```diff
@@ -10,32 +10,26 @@
 using System.Web.Http;
 using Microsoft.Azure.WebJobs.Host;
 using Newtonsoft.Json.Linq;
-using Microsoft.Azure.WebJobs.Script;
 
 namespace Microsoft.Azure.WebJobs.Script.WebHost.Handlers
 {
     public class SystemTraceHandler : DelegatingHandler
     {
         private readonly HttpConfiguration _config;
-        private TraceWriter _traceWriter;
 
         public SystemTraceHandler(HttpConfiguration config)
         {
             _config = config;
         }
 
-        private TraceWriter TraceWriter
+        internal TraceWriter TraceWriter
         {
             get
             {
-                if (_traceWriter == null)
```

Comment:
Previously we were caching the TraceWriter early on startup, meaning after we apply host.json settings, we never use the updated/configured TraceWriter. In all our controllers for example, the TraceWriter is pulled from DI on each invocation. That's what we do here now.

Reply:
Can I interpret your comment to mean that you're making this trace writing consistent with other parts of the code-base?